explainers/signed-exchange-alternate-link-explainer.md

```diff
@@ -0,0 +1,164 @@
+# Signed Exchange alternate link
+## Introduction
+We want to extend the usage of the existing [rel=alternate](https://html.spec.whatwg.org/multipage/links.html#rel-alternate) link header for signed exchange. Using this link header, the content publishers can declare that the resource is available in signed exchange format. This can be used both by the crawlers of aggregator sites  (SNS, News site, search engine..) and by the UAs. The crawlers can cache and serve the signed exchange of the content in their own server. The UAs can provide the users with a way to save the page in signe exchange format. And also signed exchange alternate links can be used to recursively prefetch appropriate subresource signed exchanges while prefetching the main resource signed exchange.
+
+## Use Cases
+### Signed Exchange discovery of main resource
+1. Content publishers can declare the URL of signed exchange format of the content using an alternate link HTTP header or using an alternate link HTML element.
+   - Example of signed exchange alternate link HTTP header in the HTTP response headers of the main resource of the content (https://publisher.example/article.html):
+      ```
+      Link: <https://sxg.publisher.example/article.html.sxg>;
+              rel="alternate";
+              type="application/signed-exchange;v=b3";
+              anchor="https://publisher.example/article.html"
+      ```
+   - Example of signed exchange alternate link HTML element in the main resource of the content (https://publisher.example/article.html):
+      ```
+      <link href="https://sxg.publisher.example/article.html.sxg"
+            rel="alternate"
+            type="application/signed-exchange;v=b3"
+            anchor="https://publisher.example/article.html">
+      ```
+1. This signed exchange alternate link of main resource can be used both by the crawlers and the UAs.
+   - When the crawlers detects the signed exchange alternate link, the crawlers can fetch the signed exchange.  And when a user is browsing the aggregator site (https://feed.example) and it has a link to the article, the signed exchange can be used for [privacy-preserving prefetching](https://wicg.github.io/webpackage/draft-yasskin-webpackage-use-cases.html#private-prefetch) using a prefetch `<link>` element and `<a>` link to the signed exchange .
+      ```
+      <link href="https://feed.example/article.html.sxg"
+            rel="prefetch">
+      <a href="https://feed.example/article.html.sxg">feed.example</a>
+      ```
+   - While the user of the UA is browsing an article (article.html), and if there is a signed exchange alternate link, the UA can provide the user with a way to save the page in signe exchange format. The saved file can be used to share with other users.
+
+### Recursive subresource signed exchange prefetch
+1. A crawler of an aggregator site fetches an article (https://publisher.example/article.html).
```

Comment:
I don't think crawlers are likely to respect subresource signed exchanges. They can just go to the original host, which can offer the signed exchange as a same-document alternative. So remove from here down to "The aggregator site will serve"?